bpf/bpf_sock.c

```diff
@@ -204,12 +211,20 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
 	struct lb4_service_v2 *svc;
 	struct lb4_key_v2 key = {
 		.dport		= ctx_get_port(ctx),
+		.address        = ctx_get_ip4(ctx),
 	};
 	struct lb4_service_v2 *slave_svc;
 
 	if (!sock_proto_enabled(ctx))
 		return CONNECT_PROCEED;
 
+	svc = __lb4_lookup_service_v2(&key);
```

Comment:
Should have a compile flag `ENABLE_EXTERNAL_SERVICES` so we don't pay the cost of looking up service twice here in fast-path if not used.